### ISSUE TITLE
Plans for future maintenance

### ISSUE THREAD

**Issue by JonathanJamesRasmussen** [Azure/iisnode#76]
Hello,
I'm currently deploying a business-critical application using iisnode on IIS 10, and was wondering if there are plans for continual maintenance on this project. I'm aware that iisnode works on IIS 10, but as updates occur, we'll need to continue to support this project, and I want to verify that iisnode will continue to work.

**Comment by mariomeyrelles** [comment]
Any news on this?

**Comment by mlhaufe** [comment]
@JonathanJamesRasmussen , @mariomeyrelles  I hope this helps:

https://twitter.com/AzureSupport/status/1153806803265654787

**Comment by mlhaufe** [comment]
FYI, Azure support has redirected me to the IIS forum:

- https://twitter.com/AzureSupport/status/1163473074563497984
- https://forums.iis.net/p/1243965/2151677.aspx?p=True&t=637018068166319507

**Comment by landgenoot** [comment]
If there was someone at Microsoft responsible for this project, we would have gotten an answer much faster than this.

So I think this answers your question.

**Comment by BryanTrach-MSFT** [comment]
@JonathanJamesRasmussen @mariomeyrelles @mlhaufe @landgenoot We apologize for the delay in reaching out to you. We checked with the product group and they verified that they have no plans to deprecate IISNode at this time. Thank you for your patience.

**Comment by landgenoot** [comment]
There is a difference between deprecated and abandoned. 

Can you be more specific about the future plans? When can we expect someone to look at the pull request that have been open for over a year? Right now, it is not actively maintained in my humble opinion.

**Comment by BryanTrach-MSFT** [comment]
@landgenoot Thank you for your concerns. I have shared them with the product group and will share any updates with you as they develop. We appreciate your patience.

**Comment by JonathanJamesRasmussen** [comment]
@BryanTrach-MSFT Thanks  for reaching out to the product group. Do you have any further information on the status of this product?

**Comment by BharathNimmala-MSFT** [comment]
@JonathanJamesRasmussen  Apologies for the delay, we are working internally with our product team to get more details on this, will keep you updated as we hear more. Thanks again for your patience.

**Comment by timothyfarmer** [comment]
@BryanTrach-MSFT Any news? Would like to know if there will be continual maintenance. Last commit was a long time ago so I'm guessing this is dying out? Bueller?

**Comment by timothyfarmer** [comment]
It's been a few months. @BryanTrach-MSFT we would love to know if this is going to be maintained by MSFT so we can decide whether this is a viable solution or if we need to move in fear of deprecation/EOL.

**Comment by NoelAbrahams** [comment]
We have been running IISNode for a number of years (5+) and it has been the weakest link in our Azure cloud infrastructure.

Decomposing an HTTP request we get the following pipeline:

* Chrome [OK] `--->`
* Azure Windows Server [OK]  `--->`
* IIS [OK]  `--->`
* IISNode [?]  `--->`
* NodeJS [OK]  `--->`
* In-house Code [That's our problem]

Currently, we are trying to investigate a performance issue where the first request is taking between 4 - 20 seconds.

Here are some UTC timings taken today:
```
Sent by Chrome at 19.56:00
Received by IIS at  19.56:00
 <IISNode> 			  <---- 14 seconds lost
Received by Node at 19:56:15 	
Returned by Node  at  19:56:15
Returned by IIS  at  19:56:15
Chrome Reported Duration =  14.74 seconds
```

We've tried all the settings in ISSNode.

I feel there is no point opening a new issue considering the last release was over 2 years ago.

@BryanTrach-MSFT, since IISNode is not deprecated, how do we get someone to investigate these issues?

Thanks

**Comment by timothyfarmer** [comment]
I've recently moved away from a node backend and adopted dotnet core with C# on the backend and just providing RESTful APIs (or GraphQL in some instances) through that platform.

**Comment by DevRCRun** [comment]
It's a little disturbing that MS seem to have problems curating their official repos. Yes, there are a lot of them, but when investigating a problem space that is precisely why it is so important to be able to quickly see the "archived" banner for stuff which is now old hat or abandoned

**Comment by drian0** [comment]
@BryanTrach-MSFT 
It is problematic to run missing critical apps on Azure App Service (Node) if one of the core components in the pipeline e.g. iisnode is not being actively maintained anymore. I came here to report a bug in iisnode but I guess what I will need to do is to find an alternative platform to host Node.

**Comment by jgwinner** [comment]
I hate to be blunt - but it sounds like IIS is going into the trashcan. Which is a shame, for all of the reasons that this project was originally created.

Node, for example - most of the millennials I work with don't even know it's single threaded. They are into SPA's and serverless, with python REST running on Lambda's. 

I don't think that invalidates SSR. I always thought IIS was a great SSR.

Yet those 14 seconds as posted by @NoelAbrahams doesn't look good.

No updates for a LONG time :(

Where is it going?

**Comment by lextm** [comment]
While the main question about IISNode's future remains unknown, you should know that HttpPlatformHandler (also from Microsoft) has been a good alternative since 2015, https://docs.lextudio.com/blog/the-rough-history-of-iis-httpplatformhandler/ And it works for both IIS and Azure App Service.

**Comment by mlsomers** [comment]
> While the main question about IISNode's future remains unknown, you should know that HttpPlatformHandler (also from Microsoft) has been a good alternative since 2015, https://docs.lextudio.com/blog/the-rough-history-of-iis-httpplatformhandler/ And it works for both IIS and Azure App Service.

Does HttpPlatformHandler also need the [UrlRewrite stuff](https://tomasz.janczuk.org/2011/08/using-url-rewriting-with-nodejs.html)? I tried configuring IISNode with `path="*"` but it seems like this will not be fixed unless someone from the community fixes it. I'll give HttpPlatformHandler a go in the mean time...

**Comment by lextm** [comment]
> Does HttpPlatformHandler also need the [UrlRewrite stuff](https://tomasz.janczuk.org/2011/08/using-url-rewriting-with-nodejs.html)?

No. You can use HttpPlatformHandler without URL rewrite module, because it uses its own internal rewrite engine to proxy requests from IIS pipeline to Node.js web server. That's why you can see that no more complex rewrite rules in the examples. @mlsomers
